### PR TITLE
[IPAM] API and first stub

### PR DESCRIPTION
**Pull request by aleoli**
# Description

This pr includes the new IPAM gRPC API

### CODE REVIEW

File: pkg/ipam/ipam.proto [hunk 1]
```diff
@@ -1,76 +1,62 @@
...
 }
 
-message SetSubnetsPerClusterRequest {
-    string remappedPodCIDR = 1;
-    string remappedExternalCIDR = 2;
-    string clusterID = 3;
+message NetworkAvailableResponse {
+    ResponseResult result = 1;
+    bool available = 2;
 }
```
Comment (fra98) [inline pkg/ipam/ipam.proto:62]: isn't the `available` boolean redundant if there is already one in the `ResponseResult`? I get that they can have different meanings, but in this case it seems redundant and not very clear

**Review by fra98 — APPROVED** [review]

File: pkg/ipam/ipam.proto (hunk 1 above)
Comment (aleoli) [reply pkg/ipam/ipam.proto:62]: This function checks if the network is available or not; in the resource result, we store the errors. We can have no errors and both available or not available networks, they are different things

Comment (fra98) [reply pkg/ipam/ipam.proto:62]: My question is more about the first boolean in the ResponseResult. What does it tell? If I am understanding correctly, I would just check if there is an error first, and then check the `available` bool, thus always discarding the first `success` bool

Comment (aleoli) [reply pkg/ipam/ipam.proto:62]: The first success is like to check `err != nil`, while this one tells you if the network is available or not

**Pull request merged**